src/main/java/net/jonathangiles/tools/teenyhttpd/util/ReflectionUtils.java

```diff
@@ -0,0 +1,60 @@
+package net.jonathangiles.tools.teenyhttpd.util;
+
+import java.lang.reflect.*;
+import java.util.Arrays;
+
+public class ReflectionUtils {
```

Comment:
If this isn't for public use, consider putting it in the `implementation` package, so it doesn't show up in javadoc, etc. Same goes for ParameterType.